### PR TITLE
firmware: Move scratch setup to _start instead of _start_warm

### LINKED ISSUE #42: Move scratch space setup to _start() from _start_warm()
In order to make _start_warm() suitable for HART hotplug, we should move scratch space setup for all HARTs in cold boot entry _start().

### PR DESCRIPTION
**Pull request by atishp04**
Scratch space setup needs to be done once for reboot for each
hart. _start_warm may be called several times if hart hotplug
is implemented.

Move scratch space setup to the beginning so that it is done
only once.

Fixes #42 
Signed-off-by: Atish Patra <atish.patra@wdc.com>

### CODE REVIEW

File: firmware/fw_base.S [hunk 1]
```diff
@@ -27,6 +27,64 @@ _start:
 
 	li	ra, 0
 	call	_reset_regs
+	/* Preload per-HART details
```
Comment (avpatel) [inline firmware/fw_base.S]: May be an empty line above this !

File: firmware/fw_base.S [hunk 2]
```diff
@@ -27,6 +27,64 @@ _start:
...
+	 * s6 -> HART ID
+	 * s7 -> HART Count
+	 * s8 -> HART Stack Size
+	 */
+	csrr	s6, CSR_MHARTID
```
Comment (avpatel) [inline firmware/fw_base.S]: The s6 is not used anywhere in the scratch_init loop. Please remove this instruction.

File: firmware/fw_base.S [hunk 3]
```diff
@@ -27,6 +27,64 @@ _start:
...
+	la	a4, platform
+#if __riscv_xlen == 64
+	lwu	s7, SBI_PLATFORM_HART_COUNT_OFFSET(a4)
+	lwu	s8, SBI_PLATFORM_HART_STACK_SIZE_OFFSET(a4)
+#else
+	lw	s7, SBI_PLATFORM_HART_COUNT_OFFSET(a4)
+	lw	s8, SBI_PLATFORM_HART_STACK_SIZE_OFFSET(a4)
+#endif
+
+	/* Setup scratch space */
```
Comment (avpatel) [inline firmware/fw_base.S]: Rephrase this to "Setup scratch space for all HARTs */

**Comment by atishp04** [comment]
Updated the PR after addressing the comments.
